subscriptions: Send update events for `is_muted` property.

laurynmm opened the pull request:
In Zulip 2.1.0, the `is_muted` stream subscription property was added and replaced the `in_home_view` property. But the server has still only been sending subscription update events with the `in_home_view` property.

Updates `do_change_subscription_property` to send a subscription update event for both `is_muted` and `in_home_view`, so that clients can fully migrate away from using `in_home_view` allowing us to eventually remove it completely.

See [relevant CZO conversation](https://chat.zulip.org/#narrow/stream/378-api-design/topic/in_home_view.20vs.20is_muted.20.28events.20only.3F.29/near/1411886).

**Notes**:

- I attempted to update `do_change_subscription_property` with the goal of making it simple to do the update to remove `in_home_view` from this function. For that final change, one should be able to remove the two `if` blocks and remove/replace all `database_property_name` / `database_value` instances to use the `property_name` and `value` parameters passed to the function.

- I thought about only sending the single event when `is_muted` was specified as the property name, but decided to always send both events so that we did not have to add extra temporary logic to `apply_event` in `events.py`. In fact, I believe that logic has been off since we haven't been sending any subscription update events with `is_muted` as the property. But since there wasn't a test in `test_events.py` for changing each subscription property name/value, it wasn't caught. Maybe a follow-up (or something to look at) would be to fill in that gap in the `test_events.py`? I did add a test for these changes and therefore for these two properties.

- Also, I made some small clean ups in the API changelog that are related to `in_home_view` or the `/update-subscription-settings` endpoint.

**Questions**:

- Did not add a **Changes** note to [/update-subscription-settings endpoint parameter](https://zulip.com/api/update-subscription-settings#parameter-subscription_data) because the change is focused on the events, but we might want to update the existing **Changes** note there for when `in_home_view` was deprecated in Zulip 2.1.0?

---

**Screenshots and screen captures:**

<details>
<summary>API changelog entry + clean-ups</summary>

![Screenshot from 2022-08-10 18-40-07](https://user-images.githubusercontent.com/63245456/183974763-30e488a1-cd71-4978-8f30-444522725d9f.png)

![Screenshot from 2022-08-10 19-35-23](https://user-images.githubusercontent.com/63245456/183979280-09d6131a-043d-46f9-ab39-f014fe284f2d.png)

![Screenshot from 2022-08-10 19-35-44](https://user-images.githubusercontent.com/63245456/183979287-9a1cc6a7-f067-4b7a-825a-96f9ce3e8a5e.png)


</details>

<details>
<summary>Update subscription event in `/get-events`</summary>

[Current documentation](https://zulip.com/api/get-events#subscription-update)
![Screenshot from 2022-08-10 18-40-40](https://user-images.githubusercontent.com/63245456/183974765-f2745937-b54c-4af0-bf2a-890b911e42d7.png)
</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented:
> Maybe a follow-up (or something to look at) would be to fill in that gap in the test_events.py?

Yeah that sounds pretty valuable -- the purpose of that test suite is to catch bugs like this.

> but we might want to update the existing Changes note there for when in_home_view was deprecated in Zulip 2.1.0?

I'm not sure it's necessary, but I suppose it couldn't hurt.

I think the main other follow-up is to open issues with the mobile and terminal apps asking them to switch to supporting the new event type; only once we've done that do we get to delete the old code.

Can you add a second commit to this PR that makes the web app use the new `is_muted` properly events rather than `in_home_view`? That's a good way to help verify end-to-end that the events API change work correctly.

timabbott commented:
This lgtm; I'll wait to merge it until we've switched the frontend to use it, since having a client implementation is a good way to manage risk of their being bugs that clients would need to work around.

laurynmm commented:
**Updates**:
- Added 2nd commit with frontend implementation.
  - Stream subscription is updated with `is_muted` event.
  - Still processing the `in_home_view` event via `update_property` in `stream_events.py`, but it does nothing. Adds a test for coverage and comments indicating that the property will be removed from the API in the future. We could simply ignore these events instead. This would raise the blueslip error/warning for an unexpected subscription property type. Wasn't sure which was better, and can update if the error would be the better option.
  - Found a bug while doing manual testing, but believe it is not related to these changes and exists currently in the app. See [this CZO conversation](https://chat.zulip.org/#narrow/stream/9-issues/topic/mute.20stream.20with.20selected.20message/near/1420246) for more details.
- In the 1st commit, added a **Changes** note to [/update-subscription-settings endpoint parameter](https://zulip.com/api/update-subscription-settings#parameter-subscription_data) as mentioned above. See screenshot below and note that it links to the specific event in the `/get-events` endpoint documentation. Again, this update doesn't feel necessary, but perhaps good prep prior to removing the property entirely.
- 1st commit is also updated for current API feature level changes that have been merged.

<details>
<summary>Screenshot of updated parameter description:</summary

[Current documentation](https://zulip.com/api/update-subscription-settings#parameters)
![Screenshot from 2022-08-15 18-58-59](https://user-images.githubusercontent.com/63245456/184891871-529856e2-7fe2-47d1-bcd1-a6e056056c93.png)
</details>

timabbott commented:
Merged, after a slight change to the comment formatting in the frontend. Thanks @laurynmm!
